src/lsd.cpp

```diff
@@ -242,7 +242,7 @@ void lsd::on_announce(udp::endpoint const& from, char const* buf
 	{
 		// we expect it to be hexadecimal
 		// if it isn't, it's not our cookie anyway
-		std::int32_t const cookie = strtol(cookie_iter->second.c_str(), nullptr, 16);
+		std::int32_t const cookie = int(strtol(cookie_iter->second.c_str(), nullptr, 16));
```

Comment:
would it make more sense to just use ``std::strtoi()``? or even better ``std::stoi()`` (which will throw an exception if the value overflows)